model/task/task.go

```diff
@@ -658,25 +660,30 @@ func GenerateNotRun() ([]Task, error) {
 		StartTimeKey:      bson.M{"$gt": time.Now().Add(-maxGenerateTimeAgo)}, // ignore older tasks, just in case
 		GenerateTaskKey:   true,                                               // task contains generate.tasks command
 		GeneratedTasksKey: bson.M{"$exists": false},                           // generate.tasks has not yet run
-		GeneratedJSONKey:  bson.M{"$exists": true},                            // config has been posted by generate.tasks command
+		"$or": []bson.M{
+			bson.M{GeneratedJSONAsStringKey: bson.M{"$exists": true}}, // config has been posted by generate.tasks command
+			bson.M{GeneratedJSONKey: bson.M{"$exists": true}},         // TODO Remove after EVG-6759 is deployed
+		},
 	}))
 }
 
 // SetGeneratedJSON sets JSON data to generate tasks from.
 func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
-	if len(t.GeneratedJSON) > 0 {
+	if len(t.GeneratedJSONAsString) > 0 || len(t.GeneratedJSON) > 0 {
 		return nil
 	}
-	t.GeneratedJSON = json
+	s := []string{}
+	for _, j := range json {
+		s = append(s, string(j))
+	}
+	t.GeneratedJSONAsString = s
 	return UpdateOne(
 		bson.M{
 			IdKey: t.Id,
-			// If this field already is set, something has gone wrong.
-			GeneratedJSONKey: bson.M{"$exists": false},
```

Comment:
Should we still check if GeneratedJSONAsStringKey already exists? It looks like this is to prevent a race.

Reply:
Fixed.

Reply:
Given what you said above about the sanity check, we only want to update if both fields are unset, not if one is unset.

Reply:
Just want to revisit this comment!

Reply:
@ablack12 Sorry it took me so long to get back to this. I don't remember my original reasoning, but my feeling now is that we would set both for a bit to make sure that this works, and then we'd stop setting `GeneratedJSONKey`. When that's the case, the race check would indeed be an or. Does that make sense?

Reply:
Yeah makes sense I think